src/pages/studyView/tabs/CNSegments.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import {observer} from "mobx-react";
+import {action, computed, observable} from "mobx";
+import autobind from "autobind-decorator";
+
+import {CopyNumberSeg} from "shared/api/generated/CBioPortalAPI";
+import IntegrativeGenomicsViewer from "shared/components/igv/IntegrativeGenomicsViewer";
+import {calcSegmentTrackHeight, defaultSegmentTrackProps, generateSegmentFeatures} from "shared/lib/IGVUtils";
+import ProgressIndicator, {IProgressIndicatorItem} from "shared/components/progressIndicator/ProgressIndicator";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import CNSegmentsDownloader from "shared/components/cnSegments/CNSegmentsDownloader";
+
+import {StudyViewPageStore, StudyViewPageTabKeyEnum} from "../StudyViewPageStore";
+
+@observer
+export default class CNSegments extends React.Component<{ store: StudyViewPageStore }, {}> {
+    @observable renderingComplete = false;
+
+    @computed get segmentTrackHeight() {
+        return calcSegmentTrackHeight(this.features);
+    }
+
+    @computed get features() {
+        const segments: CopyNumberSeg[] = this.activePromise ? this.activePromise.result || [] : [];
+
+        return generateSegmentFeatures(segments);
+    }
+
+    @computed get filename()
+    {
+        let prefix = "";
+
+        if (this.props.store.studyIds.length > 0) {
```

Comment:
Can we consolidate the logics of generating the download file name? We have a very similar logic in `clinicalDataDownloadFilename`, but the name is bit different.

Reply:
added a new computed value named `downloadFilenamePrefix`

Reply:
also added new utility functions in the new `FilenameUtils`